src/main/java/com/chnu/repository/impl/ProfileRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.chnu.repository.impl;
+
+import com.chnu.model.Profile;
+import com.chnu.repository.IProfileRepository;
+import com.chnu.util.QueryUtility;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+@Repository
+public class ProfileRepository extends AbstractRepository<Profile,Long> implements IProfileRepository {
+    @Override
+    public Optional<Profile> findByProfile(Long userId){
```

Comment:
This method isn`t needful. You can use method from AbstractRepository instead of that.